pontoon/base/tests/test_views.py

```diff
@@ -186,5 +189,96 @@ def test_given_period(self):
         """
         with patch('django.utils.timezone.now', wraps=now, return_value=aware_datetime(2015, 7, 5)):
             response = self.client.get('/contributors/?period=6')
-            assert_equal(self.mock_render.call_args[0][2]['period'], 6)
-            assert_equal(self.mock_translations_manager.call_args[0][0], aware_datetime(2015, 1, 5))
\ No newline at end of file
+            assert_equal(self.mock_render.call_args[0][0]['period'], 6)
+            assert_equal(self.mock_translations_manager.call_args[0][0], aware_datetime(2015, 1, 5))
+
+
+class ViewTestCase(TestCase):
+    def setUp(self):
+        """
+        We don't call update_projects during the tests, so we have to
+        create dummy resource project to avoid recurse redirect at /.
+        """
+        ResourceFactory.create(project=Project.objects.get(pk=1))
+
+
+class ProjectTests(ViewTestCase):
+    def test_project_doesnt_exist(self):
+        """
+        Checks if view is returning error when project slug is invalid.
+        """
+        assert_code(self.client.get('/projects/project_doesnt_exist/'), 404)
+
+    def test_project_view(self):
+        """
+        Checks if project page is returned properly.
+        """
+        project = ProjectFactory.create()
+        ResourceFactory.create(project=project)
+
+        with patch('pontoon.base.views.render', wraps=render) as mock_render:
+            response = self.client.get('/projects/{}/'.format(project.slug))
+            assert_equal(mock_render.call_args[0][2]['project'], project)
+
+
+class ProjectContributorsTests(ViewTestCase):
+    def test_project_doesnt_exist(self):
+        """
+        Checks if view handles invalid project.
+        """
+        assert_code(self.client.get('/projects/project_doesnt_exist/contributors'), 404)
+
+    def test_project_contributors(self):
+        """
+        Tests if view returns contributors spe-cific for given project.
+        """
+        project = Project.objects.get(pk=1)
```

Comment:
If possible, I prefer to use newly-created models when at all possible instead of relying on the locales and projects already in the database. It makes the tests easier to reason about since we don't have to remember what data is already in the database, and frees us up to remove or modify the pre-existing data later on.